### PR TITLE
PIN-3985 Implemented Creation of the RiskAnalisys

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**
It needs [PIN-4021](https://github.com/pagopa/interop-be-catalog-management/pull/228) and [PIN-3981](https://github.com/pagopa/interop-commons/pull/283) to compile

[PIN-3981]: https://pagopa.atlassian.net/browse/PIN-3981?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[PIN-4021]: https://pagopa.atlassian.net/browse/PIN-4021?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala [hunk 1]
```diff
@@ -662,6 +691,25 @@ object ProcessApiServiceImpl {
...
+  def isRiskAnalysisFormValid(riskAnalysisForm: Commons.RiskAnalysisForm, schemaOnlyValidation: Boolean)(
+    kind: Commons.RiskAnalysisTenantKind
+  )(implicit ec: ExecutionContext): Future[Unit] =
+    Future
+      .failed(RiskAnalysisNotValid)
+      .unlessA(
+        RiskAnalysisValidation
+          .validate(riskAnalysisForm, schemaOnlyValidation)(kind)
+          .isValid
+      )
```
Comment (beetlecrunch) [inline src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala]: No cats on Future

File: src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala [hunk 2]
```diff
@@ -662,6 +691,25 @@ object ProcessApiServiceImpl {
...
+
+  def isReceiveEService(eService: CatalogItem)(implicit ec: ExecutionContext): Future[Unit] =
+    Future.failed(EServiceNotInReceiveMode(eService.id)).unlessA(eService.mode == Receive)
+
+  def isDraftEService(eService: CatalogItem)(implicit ec: ExecutionContext): Future[Unit] =
+    Future
+      .failed(EServiceNotInDraftState(eService.id))
+      .unlessA(eService.descriptors.map(_.state) == Seq(Draft))
```
Comment (beetlecrunch) [inline src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala]: same

**Review by beetlecrunch — APPROVED** [review]

File: src/main/resources/interface-specification.yml [hunk 3]
```diff
@@ -864,6 +864,53 @@ paths:
             application/problem+json:
               schema:
                 $ref: '#/components/schemas/Problem'
+  /eservices/{eServiceId}/riskanalysis:
```
Comment (galales) [inline src/main/resources/interface-specification.yml]:
```suggestion
  /eservices/{eServiceId}/riskAnalysis:
```

File: src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala [hunk 4]
```diff
@@ -648,6 +652,31 @@ final case class ProcessApiServiceImpl(
...
+    val result = for {
+      organizationId <- getOrganizationIdFutureUUID(contexts)
+      eServiceUuid   <- eServiceId.toFutureUUID
+      catalogItem    <- catalogManagementService.getEServiceById(eServiceUuid)
+      _              <- isDraftEService(catalogItem)
+      _              <- isReceiveEService(catalogItem)
+      _              <- assertRequesterAllowed(catalogItem.producerId)(organizationId)
+      tenant         <- tenantManagementService.getTenantById(organizationId)
+      tenantKind     <- tenant.kind.toFuture(TenantKindNotFound(tenant.id))
+      _              <- isRiskAnalysisFormValid(seed.riskAnalysisForm.toTemplate, true)(tenantKind.toTemplate)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala]: very minor, but it's easier to read
```suggestion
      _              <- isRiskAnalysisFormValid(seed.riskAnalysisForm.toTemplate, schemaOnlyValidation = true)(tenantKind.toTemplate)
```

File: src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala [hunk 5]
```diff
@@ -662,6 +691,25 @@ object ProcessApiServiceImpl {
...
+  ): Future[Unit] =
+    if (
+      RiskAnalysisValidation
+        .validate(riskAnalysisForm, schemaOnlyValidation)(kind)
+        .isValid
+    ) Future.unit
+    else Future.failed(RiskAnalysisNotValid)
+
+  def isReceiveEService(eService: CatalogItem)(implicit ec: ExecutionContext): Future[Unit] =
+    Future.failed(EServiceNotInReceiveMode(eService.id)).unlessA(eService.mode == Receive)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala]: same for cats helpers on `Future`

File: src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ResponseHandlers.scala [hunk 6]
```diff
@@ -48,6 +48,21 @@ object ResponseHandlers extends AkkaResponses {
...
 
+  def createRiskAnalysisResponse[T](logMessage: String)(
+    success: T => Route
+  )(result: Try[T])(implicit contexts: Seq[(String, String)], logger: LoggerTakingImplicit[ContextFieldsToLog]): Route =
+    result match {
+      case Success(s)                             => success(s)
+      case Failure(ex: OperationForbidden.type)   => forbidden(ex, logMessage)
+      case Failure(ex: EServiceNotFound)          => notFound(ex, logMessage)
+      case Failure(ex: TenantNotFound)            => notFound(ex, logMessage)
+      case Failure(ex: TenantKindNotFound)        => notFound(ex, logMessage)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ResponseHandlers.scala]: we can avoid handling these, if they gets raised it means something very bad had happened and it's ok to return 500
```suggestion
```

**Review by galales — COMMENTED** [review]
nice PR and tests, just minor points

**Review by galales — APPROVED** [review]

**Pull request merged**